servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/NettyCompressionStreamingSerializer.java

```diff
@@ -80,8 +83,21 @@ public void onNext(@Nullable final Buffer next) {
                 }
 
                 try { // onNext will produce AT-MOST N items (as received)
-                    channel.writeInbound(extractByteBufOrCreate(next));
-                    next.skipBytes(next.readableBytes());
+                    final Buffer bufferToDecode;
+                    if (aggregate != null && aggregate.readableBytes() != 0) {
+                        aggregateNext(next);
+                        bufferToDecode = aggregate;
+                    } else {
+                        bufferToDecode = next;
+                    }
+                    ByteBuf byteBuf = toByteBuf(bufferToDecode);
+                    channel.writeInbound(byteBuf);
+                    assert byteBuf.readableBytes() == bufferToDecode.readableBytes();
+                    if (bufferToDecode.readableBytes() > 0) {
+                        aggregateRemainder(bufferToDecode);
+                    } else {
```

Comment:
`aggregate == null` is an interesting case to make sure gets tested.

Reply:
+1 on increasing test coverage. There are missing branches in `aggregateNext`, `aggregateRemainder`, and `drainChannelQueueToSingleBuffer`.